module1/homework/grayscaleImages/compression.cpp

```diff
@@ -0,0 +1,51 @@
+#include "compression.hpp" 
+
+std::vector<std::pair<uint8_t, uint8_t>> compressOneLine(const  std::array<uint8_t,width>& line){
+
+    uint8_t value;
+    uint8_t previous_value=line[0];
+    uint8_t cnt=1;
+    std::vector<std::pair<uint8_t, uint8_t>> vect;
+    auto it= line.cbegin();
+    ++it;
+    for(;it<line.cend();it++){
+        if(*it==previous_value){
+            cnt++;            
+        }
+        else{
+            vect.push_back(std::make_pair(previous_value,cnt));   
+            cnt=1;
+        } 
+        previous_value=*it;
+    }
+    if(cnt>1){
+          vect.push_back(std::make_pair(previous_value,cnt));
+          }
+          return vect;
```

Comment:
When you reconfigure it to use only once push_back. You will see that it helps you with refactoring in STL#2. Because when you only once use push_back, you can easily replace it with algorithm.

Reply:
Thanks for advice. I'll use it in STL2 homework, here I decided to leave it as it is, because I see no problems with it. Thanks